File: timer30segs/timer30segs/Form1.vb
Public Class Form1
    Dim GreenTime As Integer = 10
    Dim YelowwTime As Integer = 5
    Dim RedTime As Integer = 15
    Dim time As Integer
    Private Sub Form1_Load(sender As Object, e As EventArgs) Handles MyBase.Load
        Timer1.Enabled = True
    End Sub

    Private Sub Timer1_Tick(sender As Object, e As EventArgs) Handles Timer1.Tick
        If time = 0 Then
            Button1.BackColor = Color.Green
            Button2.BackColor = Color.Gray
            Button3.BackColor = Color.Gray
        End If
        time = time + 1

        If time >= -1 And time <= GreenTime Then
            Button1.BackColor = Color.Green
            Button2.BackColor = Color.Gray
            Button3.BackColor = Color.Gray
        ElseIf time > 10 And time <= 15 Then
            Button3.BackColor = Color.Yellow
            Button1.BackColor = Color.Gray
            Button2.BackColor = Color.Gray
        ElseIf time > 15 And time <= 30 Then
            Button2.BackColor = Color.Red
            Button1.BackColor = Color.Gray
            Button3.BackColor = Color.Gray
        End If

        If time > 30 Then
            time = 0
            Button1.BackColor = Color.Green
            Button2.BackColor = Color.Gray
            Button3.BackColor = Color.Gray
        End If

        Label1.Text = time

    End Sub
End Class

File: semaforo/semaforo/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;

namespace semaforo
{
    public partial class Form1 : Form
    {
        double time;
        bool preventivo;


        private void checkBox1_CheckedChanged(object sender, EventArgs e)
        {
            preventivo = !preventivo;
            time = 0;
        }

        public Form1()
        {
            InitializeComponent();
            timer1.Start();
        }

        private void timer1_Tick(object sender, EventArgs e)
        {
            time++;

            if (!preventivo)
            {
                if (time >= 30)
                {
                    time = 0;
                }

                if (time > 0 && time < 10)
                {
                    button1.BackColor = Color.Green;
                    button2.BackColor = Color.Black;
                    button3.BackColor = Color.Black;
                    button4.BackColor = Color.Black;
                    button5.BackColor = Color.Black;
                    button6.BackColor = Color.Red;



                }
                else if (time >= 10 && time <= 15)
                {
                    button1.BackColor = Color.Black;
                    button2.BackColor = Color.Yellow;
                    button3.BackColor = Color.Black;
                    button4.BackColor = Color.Black;
                    button5.BackColor = Color.Black;
                    button6.BackColor = Color.Red;

                }
                else if (time > 15 && time <= 25)
                {
                    button1.BackColor = Color.Black;
                    button2.BackColor = Color.Black;
                    button3.BackColor = Color.Red;
                    button4.BackColor = Color.Green;
                    button5.BackColor = Color.Black;
                    button6.BackColor = Color.Black;

                }
                else if (time > 25 && time <= 30)
                {
                    button1.BackColor = Color.Black;
                    button2.BackColor = Color.Black;
                    button3.BackColor = Color.Red;
                    button4.BackColor = Color.Black;
                    button5.BackColor = Color.Yellow;
                    button6.BackColor = Color.Black;

                }

            }
            else
            {
                if (time % 2 == 0)
                {
                    button1.BackColor = Color.Black;
                    button2.BackColor = Color.Yellow;
                    button3.BackColor = Color.Black;
                    button4.BackColor = Color.Black;
                    button5.BackColor = Color.Yellow;
                    button6.BackColor = Color.Black;

                }
                else
                {
                    button1.BackColor = Color.Black;
                    button2.BackColor = Color.Black;
                    button3.BackColor = Color.Black;
                    button4.BackColor = Color.Black;
                    button5.BackColor = Color.Black;
                    button6.BackColor = Color.Black;
                }
            }

            

        }


    }
}
